core/dbt/task/init.py

```diff
@@ -83,29 +88,196 @@ def get_addendum(self, project_name, profiles_path):
 
+    def create_profile_from_scratch(self, adapter: str):
+        """Create a profile without defaults using target_options.yml if available, or
+        sample_profiles.yml as a fallback."""
+        # Line below raises an exception if the specified adapter is not found
+        load_plugin(adapter)
+        adapter_path = get_include_paths(adapter)[0]
+        target_options_path = adapter_path / "target_options.yml"
+
+        if target_options_path.exists():
+            with open(target_options_path) as f:
+                target_options = yaml.load(f)
+            self.create_profile_from_target_options(target_options)
+        else:
+            # For adapters without a target_options.yml defined, fallback on
+            # sample_profiles.yml
+            self.create_profile_from_sample(adapter)
+
+    def check_if_can_write_profile(self, profile_name: str = None) -> bool:
+        profiles_file = os.path.join(dbt.config.PROFILES_DIR, "profiles.yml")
+        if not os.path.exists(profiles_file):
+            return True
+        profile_name = (
+            profile_name or self.get_profile_name_from_current_project()
+        )
+        with open(profiles_file, "r") as f:
+            profiles = yaml.load(f) or {}
+        if profile_name in profiles.keys():
+            response = click.confirm(
+                f"The profile {profile_name} already exists in "
+                f"{profiles_file}. Continue and overwrite it?"
+            )
+            return response
+        else:
+            return True
+
+    def create_profile_using_profile_template(self):
+        """Create a profile using profile_template.yml"""
+        with open("profile_template.yml") as f:
+            profile_template = yaml.load(f)
+        profile_name = list(profile_template["profile"].keys())[0]
+        self.check_if_can_write_profile(profile_name)
+        render_vars = {}
+        for template_variable in profile_template["vars"]:
+            render_vars[template_variable] = click.prompt(template_variable)
+        profile = profile_template["profile"][profile_name]
+        profile_str = yaml.dump(profile)
+        profile_str = Template(profile_str).render(vars=render_vars)
+        profile = yaml.load(profile_str)
+        profiles_filepath, _ = self.write_profile(profile, profile_name)
+        logger.info(
+            f"Profile {profile_name} written to {profiles_filepath} using "
+            "profile_template.yml and your supplied values."
+        )
+
+    def ask_for_adapter_choice(self) -> str:
+        """Ask the user which adapter (database) they'd like to use."""
+        click.echo("Which database would you like to use?")
+        available_adapters = list(_get_adapter_plugin_names())
+        click.echo("\n".join([
+            f"[{n+1}] {v}" for n, v in enumerate(available_adapters)
+        ]))
+        numeric_choice = click.prompt("Enter a number", type=click.INT)
+        return available_adapters[numeric_choice - 1]
```

Comment:
This could be a neat opportunity for a hint like:
```
Which database adapter would you like to use?

[1] presto
[2] postgres
[3] snowflake
[4] redshift
[5] bigquery

(Don't see the one you want? https://docs.getdbt.com/docs/available-
adapters)

Enter a number:
```